library/src/com/google/maps/android/kml/KmlContainerParser.java

```diff
@@ -0,0 +1,164 @@
+                    setContainerStyleMap(parser, containerStyleMaps);
+                } else if (parser.getName().equals(STYLE)) {
+                    setContainerStyle(parser, containerStyles);
+                } else if (parser.getName().equals(PLACEMARK)) {
+                    setContainerPlacemark(parser, containerPlacemarks);
+                } else if (parser.getName().equals(EXTENDED_DATA)) {
+                    setExtendedDataProperties(parser, containerProperties);
+                } else if (parser.getName().equals(GROUND_OVERLAY)) {
+                    containerGroundOverlays
+                            .put(KmlFeatureParser.createGroundOverlay(parser), null);
+                }
+            }
+            eventType = parser.next();
+        }
+
+        return new KmlContainer(containerProperties, containerStyles, containerPlacemarks,
+                containerStyleMaps, nestedContainers, containerGroundOverlays, containerId);
+    }
+
+    /**
+     * Creates a new style map and assigns values from the XmlPullParser parser
+     * and stores it into the container.
+     */
+    private static void setContainerStyleMap(XmlPullParser parser,
+            HashMap<String, String> containerStyleMap) throws XmlPullParserException, IOException {
+        containerStyleMap.putAll(KmlStyleParser.createStyleMap(parser));
+    }
+
+    /**
+     * Assigns properties given as an extended data element, which are obtained from an
+     * XmlPullParser and stores it in a container, Untyped <Data> only, no <SimpleData>
+     * or <Schema>, and entity replacements of the form $[dataName] are unsupported.
+     */
+    private static void setExtendedDataProperties(XmlPullParser parser,
+            HashMap<String, String> mContainerProperties)
+            throws XmlPullParserException, IOException {
+        String propertyKey = null;
+        int eventType = parser.getEventType();
+        while (!(eventType == END_TAG && parser.getName().equals(EXTENDED_DATA))) {
+            if (eventType == START_TAG) {
+                if (parser.getName().equals("Data")) {
+                    propertyKey = parser.getAttributeValue(null, "name");
+                } else if (parser.getName().equals("value") && propertyKey != null) {
+                    mContainerProperties.put(propertyKey, parser.nextText());
+                    propertyKey = null;
+                }
+            }
+            eventType = parser.next();
+        }
+    }
+
+    /**
+     * Creates a new default Kml Style with a specified ID (given as an attribute value in the
+     * start tag) and assigns specific elements read from the XmlPullParser to the Style. A new
+     * style is not created if it does not have an ID.
+     */
+    private static void setContainerStyle(XmlPullParser parser,
+            HashMap<String, KmlStyle> containerStyles) throws XmlPullParserException, IOException {
+        Boolean hasStyleId = parser.getAttributeValue(null, "id") != null;
```

Comment:
You can in-line this check and remove the hasStyleId variable. As a bonus it removes the Boolean boxed type, which you don't need here.